### PR TITLE
Des 157 make functional the olvide mi contrasena option adding email pashe 2 cms

### PR DESCRIPTION
**Pull request by Daniielmayo**

### CODE REVIEW

File: controllers/auth.js [hunk 1]
```diff
@@ -1,193 +1,242 @@
...
+      country: user.country,
+      city: user.city,
+      phone: user.phone,
+      token,
+    });
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({
+      ok: false,
+      msg: "Por favor hable con el administrador",
```
Comment (DanielCalleCO) [inline controllers/auth.js:56]:
```suggestion
      msg: "Por favor comunícate con el administrador",
```

File: controllers/auth.js [hunk 2]
```diff
@@ -1,193 +1,242 @@
...
-//todo: renew token
+    //* confirm the passwords
+    const validPassword = bcrypt.compareSync(password, user.password);
 
-const revalidateToken = async (req, res = response) => {
-    const { name, email, uid, city, lastname, phone, country } = req;
+    if (!validPassword) {
+      return res.status(400).json({
+        ok: false,
+        msg: "Password incorrecto",
```
Comment (DanielCalleCO) [inline controllers/auth.js:82]:
```suggestion
        msg: "Contraseña incorrecta",
```

File: controllers/auth.js [hunk 3]
```diff
@@ -1,193 +1,242 @@
 const { response } = require("express");
 const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
```
Comment (abdulflorez) [inline controllers/auth.js:3]: What about changing this for something more clear,  jsonToken

File: controllers/auth.js [hunk 4]
```diff
@@ -1,193 +1,242 @@
...
+      country: user.country,
+      phone: user.phone,
+
+      token,
     });
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({
+      ok: false,
+      msg: "Por favor hable con el administrador",
```
Comment (DanielCalleCO) [inline controllers/auth.js:114]:
```suggestion
      msg: "Por favor comunícate con el administrador",
```

File: controllers/auth.js [hunk 5]
```diff
@@ -1,193 +1,242 @@
...
+    res.json({
+      ok: true,
+      uid: user.id,
+      token,
+    });
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({
+      ok: false,
+      msg: "Por favor hable con el administrador",
```
Comment (DanielCalleCO) [inline controllers/auth.js:231]:
```suggestion
      msg: "Por favor comunícate con el administrador",
```

File: middlewares/validate-email-reset-password.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    },
+  });
+
+  let user = await User.findOne({ email });
+  const resetLink = `http://localhost:3000/resetPassword/${user.id}/${resetToken}`;
+  await transporter.sendMail({
+    from: user.email, //user email
+    to: email,
+    subject: "Restablecimiento de Contraseña",
+    html: `¡Hola! Haga clic en el siguiente enlace para restablecer su contraseña: <br/>
```
Comment (DanielCalleCO) [inline middlewares/validate-email-reset-password.js:19]:
```suggestion
    html: `¡Hola! Haz clic en el siguiente enlace para restablecer tu contraseña: <br/>
```

File: middlewares/validate-email-reset-password.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  const transporter = nodemailer.createTransport({
+    service: "Gmail",
+    auth: {
+      user: "pruebasdanielmayo@gmail.com",
+      pass: "tfehpymjjkfwsagn",
+    },
+  });
+
+  let user = await User.findOne({ email });
+  const resetLink = `http://localhost:3000/resetPassword/${user.id}/${resetToken}`;
```
Comment (DanielCalleCO) [inline middlewares/validate-email-reset-password.js:14]: local host

**Review by abdulflorez — CHANGES_REQUESTED** [review]
I wrote some comments

**Review by DanielCalleCO — CHANGES_REQUESTED** [review]
Hello Dani, could you please review my suggestions?

**Review by abdulflorez — DISMISSED** [review]

File: middlewares/validate-email-reset-password.js [hunk 8]
```diff
@@ -11,12 +11,12 @@ const sendPasswordResetEmail = async (email, resetToken) => {
...
 
   let user = await User.findOne({ email });
-  const resetLink = `http://localhost:3000/resetPassword/${user.id}/${resetToken}`;
+  const resetLink = `https://regresemos-a-nuestra-casa-ui-hajm.vercel.app/resetPassword/${user.id}/${resetToken}`;
   await transporter.sendMail({
     from: user.email, //user email
     to: email,
     subject: "Restablecimiento de Contraseña",
-    html: `¡Hola! Haga clic en el siguiente enlace para restablecer su contraseña: <br/>
+    html: `¡Hola! esperamos que te encuentres bien. Hemos recibido una solicitud para recuperar la contraseña de tu cuenta  Entendemos lo importante que es para ti acceder a tu cuenta de manera segura.  <br/>Haz clic en el siguiente enlace para restablecer tu contraseña: <br/>
```
Comment (DanielCalleCO) [inline middlewares/validate-email-reset-password.js:19]:
```suggestion
    html: `¡Hola! esperamos que te encuentres bien. <br/>Hemos recibido una solicitud para recuperar tu contraseña. Entendemos lo importante que es para ti acceder a tu cuenta de manera segura.  <br/>Haz clic en el siguiente enlace para restablecer tu contraseña: <br/>
```

**Review by DanielCalleCO — CHANGES_REQUESTED** [review]
Nice! Just a simple additional comment before merge...

File: middlewares/validate-email-reset-password.js [hunk 9]
```diff
@@ -16,7 +16,7 @@ const sendPasswordResetEmail = async (email, resetToken) => {
     from: user.email, //user email
     to: email,
     subject: "Restablecimiento de Contraseña",
-    html: `¡Hola! esperamos que te encuentres bien. Hemos recibido una solicitud para recuperar la contraseña de tu cuenta  Entendemos lo importante que es para ti acceder a tu cuenta de manera segura.  <br/>Haz clic en el siguiente enlace para restablecer tu contraseña: <br/>
+    html: `¡Hola! esperamos que te encuentres bien. <br/>Hemos recibido una solicitud para recuperar tu contraseña. Entendemos lo importante que es para ti acceder a tu cuenta de manera segura.  <br/>Haz clic en el siguiente enlace para restablecer tu contraseña: <br/>
```
Comment (DanielCalleCO) [inline middlewares/validate-email-reset-password.js:19]: Sorry I forgot to let you the last suggestion.

```suggestion
    html: `¡Hola! esperamos que te encuentres bien. <br/>Hemos recibido una solicitud para recuperar tu contraseña. Entendemos lo importante que es para ti acceder a tu cuenta de manera segura.  <br/>Haz clic en el siguiente enlace para restablecerla: <br/>
```

**Review by DanielCalleCO — CHANGES_REQUESTED** [review]

**Review by DanielCalleCO — APPROVED** [review]

**Review by abdulflorez — APPROVED** [review]
Lgtm

**Pull request merged**